### ISSUE TITLE
Addon blocks item upgrading

### ISSUE THREAD

**Issue by CrenshawCoding** [0xbs/premade-groups-filter#112]
When trying to upgrade an item, I get the following error sometimes: 
```
1x [ADDON_ACTION_FORBIDDEN] AddOn 'PremadeGroupsFilter' tried to call the protected function 'UpgradeItem()'.
[string "@Interface/AddOns/!BugGrabber/BugGrabber.lua"]:480: in function <Interface/AddOns/!BugGrabber/BugGrabber.lua:480>
[string "=[C]"]: in function `UpgradeItem'
[string "@Interface/AddOns/Blizzard_ItemUpgradeUI/Blizzard_ItemUpgradeUI.lua"]:104: in function `OnConfirm'
[string "@Interface/FrameXML/StaticPopup.lua"]:355: in function `OnAccept'
[string "@Interface/FrameXML/StaticPopup.lua"]:5269: in function `StaticPopup_OnClick'
[string "*StaticPopup.xml:24_OnClick"]:1: in function <[string "*StaticPopup.xml:24_OnClick"]:1>
```
Steps to reproduce:
1. Try to upgrade an item (in my case with valor)

**Comment by 0xbs** [comment]
I just released a fix that should reduce the number of UI taints. A complete fix will need more time.

**Comment by 0xbs** [comment]
Should be completely fixed in version 3.2.0
